modAL/models/base.py

```diff
@@ -250,29 +165,16 @@ def query(self, X_pool, *query_args, **query_kwargs) -> Union[Tuple, modALinput]
             labelled and the instances themselves. Can be different in other cases, for instance only the instance to be
             labelled upon query synthesis.
         """
-        query_result = self.query_strategy(self, X_pool, *query_args, **query_kwargs)
+        query_result, query_metrics = self.query_strategy(self, X_pool, *query_args, **query_kwargs)
 
         if isinstance(query_result, tuple):
             warnings.warn("Query strategies should no longer return the selected instances, "
                           "this is now handled by the query method. "
                           "Please return only the indices of the selected instances.", DeprecationWarning)
             return query_result
 
-        return query_result, retrieve_rows(X_pool, query_result)
-
-    def score(self, X: modALinput, y: modALinput, **score_kwargs) -> Any:
-        """
-        Interface for the score method of the predictor.
-
-        Args:
-            X: The samples for which prediction accuracy is to be calculated.
-            y: Ground truth labels for X.
-            **score_kwargs: Keyword arguments to be passed to the .score() method of the predictor.
+        return query_result, retrieve_rows(X_pool, query_result), query_metrics
```

Comment:
Hi @cosmic-cortex, 

we had another idea:

We changed the behavior of all functions in selection.py (shuffled_argmax , multi_argmax, ... ), so that they do not only return the indices but now also the corresponding values. 

With these adoptions, all query strategies do return now the metrics. 
The decision if the metrics should be returned can then be made in the query methods. 

What do you think about this?